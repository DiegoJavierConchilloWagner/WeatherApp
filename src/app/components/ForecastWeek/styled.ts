import styled from 'styled-components';

export const StyledForecastWeek = styled.div`
	display: flex;
	flex-wrap: wrap;
	margin: 3rem 10vw;
	justify-content: center;
	gap: 2rem;
	@media (max-width: 768px) {
		gap: 0%;
	}
	::-webkit-scrollbar {
		height: 7px;
		width: 2px;
		background: ${({ theme }) => theme.colors.background.color};
	}
	::-webkit-scrollbar-thumb:horizontal {
		background: #83c2da;
		border-radius: 25px;
	}
	background: rgba(255, 255, 255, 0.1);
	box-shadow: 0 8px 32px 0 rgba(31, 38, 135, 0.37);
	backdrop-filter: blur(0px);
	-webkit-backdrop-filter: blur(0px);
	border-radius: 10px;
	border: 1px solid rgba(255, 255, 255, 0.18);
`;

export const StyledForecastedDay = styled.div`
	scroll-snap-align: start;
	padding: 1rem;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	color: ${({ theme }) => theme.colors.typography.primary};
	font-family: ${({ theme }) => theme.fonts.primary};
	text-align: center;

	@media (max-width: 768px) {
		:not(:last-child) {
			border-bottom: 1px solid white;
		}
	}

	h4 {
		font-weight: 400;
	}
`;

export const IconContainer = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
`;

export const Icon = styled.div`
	flex-direction: column;

	h5 {
		font-weight: 500;
		text-align: center;
	}
`;

export const TemperaturesContainer = styled.div`
	display: flex;
	flex-direction: column;
`;
