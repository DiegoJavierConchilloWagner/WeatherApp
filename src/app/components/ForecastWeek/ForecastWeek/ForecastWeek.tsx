import { FC } from 'react';

import { useSelector } from 'react-redux';

import { IWeekMapped } from '@common/index';

import { RootState } from '@redux/store/store';
import { ForecastDay } from '../ForecastDay/ForecastDay';
import { StyledForecastWeek } from '../styled';

interface Props {
	isLoading: boolean;
}

export const ForecastWeek: FC<Props> = ({ isLoading }) => {
	const { week } = useSelector((state: RootState) => state.WeatherInterface);

	return (
		<StyledForecastWeek>
			{week.map(
				(day: IWeekMapped, index: number) =>
					index > 0 && <ForecastDay key={day.dt} day={day} isLoading={isLoading} />
			)}
		</StyledForecastWeek>
	);
};
