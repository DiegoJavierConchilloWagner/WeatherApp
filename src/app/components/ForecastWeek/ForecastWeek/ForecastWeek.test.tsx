import renderWithThemeAndProvider from '@common/utils/renderWhitThemeAndProvider';
import { ForecastWeek } from './ForecastWeek';

describe('Example ForecastSwitch', () => {
	it('should render correcty', () => {
		const wrapper = renderWithThemeAndProvider(<ForecastWeek isLoading={false} />);

		expect(wrapper).toMatchSnapshot();
	});
});
