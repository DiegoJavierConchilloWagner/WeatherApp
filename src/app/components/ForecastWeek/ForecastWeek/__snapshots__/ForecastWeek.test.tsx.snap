// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example ForecastSwitch should render correcty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 3rem 10vw;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2rem;
  background: rgba(255,255,255,0.1);
  box-shadow: 0 8px 32px 0 rgba(31,38,135,0.37);
  -webkit-backdrop-filter: blur(0px);
  backdrop-filter: blur(0px);
  -webkit-backdrop-filter: blur(0px);
  border-radius: 10px;
  border: 1px solid rgba(255,255,255,0.18);
}

.c0::-webkit-scrollbar {
  height: 7px;
  width: 2px;
  background: #0d0f40;
}

.c0::-webkit-scrollbar-thumb:horizontal {
  background: #83c2da;
  border-radius: 25px;
}

@media (max-width:768px) {
  .c0 {
    gap: 0%;
  }
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 3rem 10vw;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2rem;
  background: rgba(255,255,255,0.1);
  box-shadow: 0 8px 32px 0 rgba(31,38,135,0.37);
  -webkit-backdrop-filter: blur(0px);
  backdrop-filter: blur(0px);
  -webkit-backdrop-filter: blur(0px);
  border-radius: 10px;
  border: 1px solid rgba(255,255,255,0.18);
}

.c0::-webkit-scrollbar {
  height: 7px;
  width: 2px;
  background: #0d0f40;
}

.c0::-webkit-scrollbar-thumb:horizontal {
  background: #83c2da;
  border-radius: 25px;
}

@media (max-width:768px) {
  .c0 {
    gap: 0%;
  }
}

<div>
    <div
      class="c0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
